automation/taskcluster/decision_task_publish.py

```diff
@@ -0,0 +1,224 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+Decision task for releases
+"""
+
+from __future__ import print_function
+import argparse
+import json
+import os
+import taskcluster
+import sys
+
+import lib.module_definitions
+import lib.tasks
+import lib.util
+
+TASK_ID = os.environ.get('TASK_ID')
+HEAD_REV = os.environ.get('MOBILE_HEAD_REV')
+BRANCH = os.environ.get('MOBILE_HEAD_BRANCH')
+REPO_URL = os.environ.get('MOBILE_HEAD_REPOSITORY')
+
+BUILDER = lib.tasks.TaskBuilder(
+    task_id=TASK_ID,
+    repo_url=os.environ.get('MOBILE_HEAD_REPOSITORY'),
+    branch=BRANCH,
+    commit=HEAD_REV,
+    owner="skaspari@mozilla.com",
+    source='{}/raw/{}/.taskcluster.yml'.format(REPO_URL, HEAD_REV),
+    scheduler_id=os.environ.get('SCHEDULER_ID'),
+)
+
+BUILD_GRADLE_TASK_NAMES = ('assemble', 'test', 'lint')
+
+
+def _get_gradle_module_name(name):
+    return ':{}'.format(name)
+
+
+def generate_build_task(version, artifact_info, is_snapshot, is_staging):
+    checkout = ("export TERM=dumb && git fetch {} {} --tags && "
+                "git config advice.detachedHead false && "
+                "git checkout {}".format(REPO_URL, BRANCH, HEAD_REV))
+
+    artifacts = {
+        artifact_info['artifact']: {
+            'type': 'file',
+            'expires': taskcluster.stringDate(taskcluster.fromNow('1 year')),
+            'path': artifact_info['path']
+        }
+    }
+
+    snapshot_flag = '-Psnapshot ' if is_snapshot else ''
+    module_name = _get_gradle_module_name(artifact_info['name'])
+    gradle_tasks_for_this_module_only = ' '.join(
+        '{}:{}'.format(module_name, gradle_task)
+        for gradle_task in BUILD_GRADLE_TASK_NAMES
+    )
+
+    return taskcluster.slugId(), BUILDER.build_task(
+        name='Android Components - Module {} ({})'.format(module_name, version),
+        description='Building and testing module {}'.format(module_name),
+        command=(
+            checkout +
+            ' && ./gradlew --no-daemon clean ' +
+            snapshot_flag + gradle_tasks_for_this_module_only +
+            ' docs uploadArchives zipMavenArtifacts'
+        ),
+        features={
+            'chainOfTrust': True,
+        },
```

Comment:
Nice to see the formerly `taskclusterProxy` off from this list of features, nice cleanup!